x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -225,27 +225,33 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
         }
 
         await ruleDataService.createOrUpdateComponentTemplate({
-          name: componentTemplateName,
-          body: {
-            template: {
-              settings: {
-                number_of_shards: 1,
+          template: {
+            name: componentTemplateName,
+            body: {
+              template: {
+                settings: {
+                  number_of_shards: 1,
+                },
+                mappings: {}, // TODO: Add mappings here via `mappingFromFieldMap()`
               },
-              mappings: {}, // TODO: Add mappings here via `mappingFromFieldMap()`
             },
           },
+          templateVersion: 1,
         });
 
         await ruleDataService.createOrUpdateIndexTemplate({
-          name: ruleDataService.getFullAssetName('security-solution-index-template'),
-          body: {
-            index_patterns: [ruleDataService.getFullAssetName('security-solution*')],
-            composed_of: [
-              ruleDataService.getFullAssetName(TECHNICAL_COMPONENT_TEMPLATE_NAME),
-              ruleDataService.getFullAssetName(ECS_COMPONENT_TEMPLATE_NAME),
-              componentTemplateName,
-            ],
+          template: {
+            name: ruleDataService.getFullAssetName('security-solution-index-template'),
+            body: {
+              index_patterns: [ruleDataService.getFullAssetName('security-solution*')],
```

Comment:
Probably this is just a leftover from some previous draft, but in https://github.com/elastic/kibana/issues/98912 we decided that the index containing detection alerts would be called `security.alerts`, not `security-solution`.

Btw @xcrzx will need to set up the 2nd `security.events` index for the rule execution log.